EmotiBitOscilloscope/src/ofApp.cpp

```diff
@@ -1171,6 +1236,8 @@ void ofApp::clearOscilloscopes()
 	for (int w = 0; w < scopeWins.size(); w++) {
 		scopeWins.at(w).clearData();
 	}
+	
+	oscilloscopeStreamCountUpdated = false;
```

Comment:
No, the `updateAvailableDataStreams` function handles addition requests, if it has already been added.

But, i think a better way to handle this might be to pass a bool into `clearOscilloscopes` , something like `clearOscilloscopes(bool newDeviceSelected)`, which controls `oscilloscopeStreamCountUpdated`. This way, i can also move `updateAvailableDataStreams(EmotiBitPacket::TypeTag::THERMOPILE, false)` inside `clearOscilloscopes`